src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/WarningCaseEventHandler.java

```diff
@@ -63,34 +67,98 @@ public WarningCaseEventHandler(AuthTokenGenerator serviceAuthGenerator, Workflow
         return response.getResults();
     }
 
+    @SuppressWarnings("PMD.ConfusingTernary")
     @Override
     public void handle(List<? extends EvaluateResponse> results, EventInformation eventInformation) {
+        Set<CancellationEvaluateResponse> emptyWarnings = new LinkedHashSet<>();
+        Set<CancellationEvaluateResponse> ctgWarnings = new LinkedHashSet<>();
+        Set<Warning> warnings = new LinkedHashSet<>();
+
         results.stream()
             .filter(CancellationEvaluateResponse.class::isInstance)
             .map(CancellationEvaluateResponse.class::cast)
             .filter(result -> CancellationActions.WARN == CancellationActions.from(result.getAction().getValue()))
-            .forEach(cancellationEvaluateResponse -> {
-                DmnValue<String> taskCategories = cancellationEvaluateResponse.getTaskCategories();
-                DmnValue<String> processCategories = cancellationEvaluateResponse.getProcessCategories();
+            .forEach(warnResponse -> {
+                if (StringUtils.isEmpty(warnResponse.getWarningCode())
+                    || StringUtils.isEmpty(warnResponse.getWarningText() != null)) {
```

Comment:
As per Sonar -> For this part of new logic, can we introduce the unit test for  alternate condition